apps/web/src/app/(private)/evaluations/[evaluationUuid]/editor/_components/EvaluationEditor/Editor/Playground/index.tsx

```diff
@@ -0,0 +1,170 @@
+'use client'
+
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+import { ConversationMetadata } from '@latitude-data/compiler'
+import { EvaluationDto } from '@latitude-data/core/browser'
+import {
+  Badge,
+  Button,
+  Icons,
+  Input,
+  Text,
+  TextArea,
+} from '@latitude-data/web-ui'
+import { ROUTES } from '$/services/routes'
+import useProviderLogs from '$/stores/providerLogs'
+import Link from 'next/link'
+import { useSearchParams } from 'next/navigation'
+
+import { Header } from '../Header'
+import Preview from './Preview'
+
+type InputDto = {
+  value: string
+  type: 'textarea' | 'input'
+}
+
+function convertParams(inputs: Record<string, InputDto>) {
+  return Object.fromEntries(
+    Object.entries(inputs).map(([key, inputDto]) => {
+      let value
+      try {
+        value = JSON.parse(inputDto.value)
+      } catch (e) {
+        // Do nothing
+      }
+      return [key, value]
+    }),
+  )
+}
+
+export default function Playground({
+  evaluation,
+  metadata,
+}: {
+  evaluation: EvaluationDto
+  metadata: ConversationMetadata
+}) {
+  const [mode, setMode] = useState<'preview' | 'chat'>('preview')
+  const [inputs, setInputs] = useState<Record<string, InputDto>>({})
+  const parameters = useMemo(() => convertParams(inputs), [inputs])
+  const searchParams = useSearchParams()
+  const providerLogUuid = searchParams.get('providerLogUuid')
+  const { data } = useProviderLogs()
+  const providerLog = useMemo(
+    () => data?.find((log) => log.uuid === providerLogUuid),
+    [data, providerLogUuid],
+  )
+
+  const setInput = useCallback((param: string, value: InputDto) => {
+    setInputs((inputs) => ({ ...inputs, [param]: value }))
+  }, [])
+
+  useEffect(() => {
+    console.log(providerLog)
+
+    if (providerLog) {
+      setInput('messages', {
+        type: 'textarea',
+        value: JSON.stringify(providerLog.messages),
+      })
+    }
+  }, [setInput, providerLog])
+
+  useEffect(() => {
+    if (!metadata) return
+
+    // Remove only inputs that are no longer in the metadata, and add new ones
+    // Leave existing inputs as they are
+    setInputs(
+      Object.fromEntries(
+        Array.from(metadata.parameters).map((param) => {
+          if (param in inputs) return [param, inputs[param]!]
+
+          return [param, { value: '', type: 'input' }]
+        }),
+      ),
+    )
+  }, [metadata])
```

Comment:
There cannot be "new inputs". Evaluation only have the default predefined parameters that are extracted from a log.

Reply:
hmm but the user can edit the evaluation prompt and remove the predefined parameters and add them back in

Reply:
Yeah, but the parameters always exist, wether you use them in the prompt or not. It is not like Documents, where ONLY when you reference a variable in the code a new parameter is created. In this case, users must only reference variables from a list of predefined variables, which are filled with data automatically from logs. I think we should always display this list to the user, so they’re aware of the parameters they have access to, even if they’re not currently using them in their code.

Reply:
hmm weird ux imo, I'd rather show them a message telling them they have to include the messages variable if it's not present in the document, rather than showing an input for a variable that they have removed from the document

Reply:
Yeah, but the parameters always exist, wether you use them in the prompt or not. It is not like Documents, where ONLY when you reference a variable in the code a new parameter is created. In this case, users must only reference variables from a list of predefined variables, which are filled with data automatically from logs. I think we should always display this list to the user, so they’re aware of the parameters they have access to, even if they’re not currently using them in their code.

(resending this message, this was removed by accident)

Reply:
but if they don't use it in the document they won't have an effect at all when you run the prompt no? it's just weird to have an input for a variable that won't have any usage in the document. I understand that internally they exist in the state of the compiler when running the document, but that's not relevant to the user. It's better to tell the user "hey, remember to add the messages variable to your document" rather than just showing an input that has no effect on the evaluation

Reply:
Maybe we shouldn't display it as a bunch of inputs. We could just display a "log" there, with the ability to modify it.